tests/tls_e2e/client_enc/openssl_client.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/edger8r/enclave.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/syscall/arpa/inet.h>
+#include <openenclave/internal/syscall/device.h>
+#include <openenclave/internal/syscall/netdb.h>
+#include <openenclave/internal/syscall/netinet/in.h>
+#include <openenclave/internal/syscall/sys/socket.h>
+#include <openenclave/internal/tests.h>
+
+#include <arpa/inet.h>
+#include <errno.h> // For errno & error defs
+#include <netdb.h>
+#include <netinet/in.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <openssl/evp.h>
+#include <openssl/ssl.h>
+
+#include "../common/openssl_utility.h"
+#include "tls_e2e_t.h"
+
+extern "C"
+{
+    int launch_tls_client(
+        struct tls_control_args* config,
+        char* server_name,
+        char* server_port);
+    int setup_tls_server(struct tls_control_args* config, char* server_port);
+};
+
+struct tls_control_args g_control_config;
+
+// create a socket and connect to the server_name:server_port
+int create_socket(int& client_socket)
+{
+    int ret = -1;
+    struct sockaddr_in ipOfServer;
+    if ((client_socket = socket(AF_INET, SOCK_STREAM, 0)) < 0)
+    {
+        OE_TRACE_ERROR("Socket not created \n");
+        goto exit;
+    }
+
+    ipOfServer.sin_family = AF_INET;
+    ipOfServer.sin_port = htons(12345);
+    ipOfServer.sin_addr.s_addr = inet_addr("127.0.0.1");
+
+    if (connect(
+            client_socket, (struct sockaddr*)&ipOfServer, sizeof(ipOfServer)) <
+        0)
+    {
+        OE_TRACE_ERROR("Connection failed due to port and ip problems\n");
```

Comment:
For the consistency, can you double check whether we should add `TLS_CLIENT` or `TLS_SERVER` to the log?